File: project/research/general_n/2026-09-17-d5-exact-cover-v1/check_d5_cover.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
 vector<pair<int,int>> E; for(int i=0;i<5;i++)for(int j=i+1;j<5;j++)E.push_back({i,j});
 int best=99; map<pair<int,int>,int> hist; map<string,int> types;
 for(int mask=0;mask<(1<<10);++mask){
  vector<int> qidx(10,-1); vector<pair<int,int>> Q; int deg[5]={};
  for(int b=0;b<10;b++)if(mask>>b&1){qidx[b]=Q.size();Q.push_back(E[b]);deg[E[b].first]++;deg[E[b].second]++;}
  int m=Q.size(), full=(1<<m)-1; vector<pair<int,int>> fam;
  for(int S=1;S<32;S++){
   int r=__builtin_popcount((unsigned)S), cost=(r==5?0:4-r), bm=0;
   for(int qi=0;qi<m;qi++){
    auto [i,j]=Q[qi]; bool threat=false;
    for(auto [t,s]: {pair<int,int>{i,j},pair<int,int>{j,i}}){
      if((S>>t&1)||!(S>>s&1)) continue;
      int cnt=0; for(int v=0;v<5;v++) if((S>>v&1) && v!=t){
        int a=min(t,v),b=max(t,v), bi=-1; for(int k=0;k<10;k++) if(E[k]==make_pair(a,b)){bi=k;break;}
        if(bi>=0 && (mask>>bi&1))cnt++;
      }
      if(cnt==1) threat=true;
    }
    if(threat) bm|=1<<qi;
   }
   if(bm)fam.push_back({bm,cost});
  }
  const int INF=99; vector<int> dp(1<<m,INF); dp[0]=0;
  for(int st=0;st<(1<<m);st++) if(dp[st]<INF) for(auto [bm,c]:fam) dp[st|bm]=min(dp[st|bm],dp[st]+c);
  int c=dp[full], mu=10-m, total=2*mu+c; best=min(best,total); hist[{mu,c}]++;
  sort(deg,deg+5); if(total==11){ string k=to_string(mu)+":"+to_string(c)+":"; for(int x:deg)k+=to_string(x);types[k]++; }
  cout<<mask<<' '<<mu<<' '<<c<<' '<<total; for(int x:deg)cout<<' '<<x; cout<<'\n';
 }
 cerr<<"best="<<best<<"\n"; for(auto &[k,v]:types)cerr<<k<<" "<<v<<"\n";
}

File: project/research/general_n/2026-09-17-d5-exact-cover-v1/check_d5_cover.py
import itertools, json, hashlib
D=5
verts=range(D)
edges=list(itertools.combinations(verts,2))

def B(mask,S):
    S=set(S); adj=[set() for _ in verts]; Q=[]
    for b,(i,j) in enumerate(edges):
        if mask>>b&1:
            adj[i].add(j);adj[j].add(i);Q.append((i,j))
    out=set()
    for i,j in Q:
        for t,s in ((i,j),(j,i)):
            if t not in S and s in S and (adj[t]&S)=={s}:
                out.add((min(i,j),max(i,j)))
    return out

def mincost(mask):
    Q=[e for b,e in enumerate(edges) if mask>>b&1]
    if not Q: return 0
    idx={e:i for i,e in enumerate(Q)}; full=(1<<len(Q))-1
    fam=[]
    for r in range(1,6):
        cost=0 if r==5 else 4-r
        for S in itertools.combinations(verts,r):
            bm=0
            for e in B(mask,S): bm |= 1<<idx[e]
            if bm: fam.append((bm,cost,S))
    INF=99; dp=[INF]*(1<<len(Q));dp[0]=0
    for s in range(1<<len(Q)):
        if dp[s]>=INF: continue
        for bm,c,_ in fam:
            ns=s|bm
            if dp[s]+c<dp[ns]:dp[ns]=dp[s]+c
    return dp[full]

rows=[]; hist={}; types={}; best=99
for mask in range(1<<10):
    mu=10-mask.bit_count(); c=mincost(mask); total=2*mu+c
    deg=[0]*5
    for b,(i,j) in enumerate(edges):
        if mask>>b&1:deg[i]+=1;deg[j]+=1
    rows.append((mask,mu,c,total,*sorted(deg)))
    hist.setdefault(mu,{}); hist[mu][c]=hist[mu].get(c,0)+1
    best=min(best,total)
for row in rows:
    if row[3]==best:
        key=(row[1],row[2],tuple(row[4:]));types[str(key)]=types.get(str(key),0)+1
text=''.join(' '.join(map(str,r))+'\n' for r in rows).encode()
summary={'status':'PASS_EXHAUSTIVE_D5_SUPPORT_COVER','tight_graphs':1024,'minimum_total_defect':best,'histogram_by_mu_and_cover_cost':hist,'minimum_types':types,'sha256':hashlib.sha256(text).hexdigest()}
print(json.dumps(summary,indent=2,sort_keys=True))
